coalib/core/Bear.py

```diff
@@ -130,6 +132,21 @@ class Bear:
     CAN_FIX = set()
     ASCIINEMA_URL = ''
     BEAR_DEPS = set()
+    """
+        Attributes for a new bear.
+
+        :param section:
+            The section object where bear settings are contained.
+        :param file_dict:
+            The file-dictionary containing a mapping of filenames to the
+            according file contents.
+        :raises RuntimeError:
+            Raised when bear requirements are not fulfilled.
+    """
+    section = attr.ib(hash=False)
+    file_dict = attr.ib(hash=False)
+    _dependency_results = attr.ib(init=False, default=defaultdict(list),
+                                  hash=False)
```

Comment:
for `repr` we have thankfully `@generate_repr` :D

> ... and can even compare bears with __le__ and the like.

The problem is: it makes no sense to compare them, except `==` (though we need to be also careful with that, the new core defines whether considering two bears equal, this plays a large role in the bear-dependency system). The default instance check is the safest currently imo^^

Many other classes would really benefit from `attr.s`, though it's a bit tricky for this one. It's used in multiple processes and contains partially unhashable content^^